scripts/dashboard/validate_dashboards_format.py

```diff
@@ -0,0 +1,153 @@
+      }
+
+    if pathParts[3].split('.')[-1] == "json":
+      map[pathParts[2]]['jsonFiles'].append(pathParts[3])
+
+  return map
+
+def get_sample_dashboards_json(path):
+  '''Checks if file is in json format and returns data in dictionary form'''
+  with open(path) as f:
+    try:
+      dashboards_dict = json.load(f)
+      dashboards_json = json.dumps(dashboards_dict)
+
+      return dashboards_dict
+
+    except JsonFormattingError:
+      print("{} content could not be loaded".format(path))
+
+def check_json_file_name(path, file_name_parts):
+  '''Checks if json file name is in the proper format'''
+  if len(file_name_parts) != 2:
+    raise JsonFormattingError("{} file name not in <name>.json format".format(path))
+
+def mapJsonFiles(directory, pathMap):
+  '''Stores json data into dictionary'''
+  for jsonFile in pathMap[directory]['jsonFiles']:
+    jsonPath = os.path.join('.', 'dashboards', directory, jsonFile)
+    dashDict = get_sample_dashboards_json(jsonPath)
+
+    if not 'displayName' in dashDict.keys():
+      raise JsonFormattingError("{} is missing displayName field".format(jsonFile))
+
+    pathMap[directory]['jsonNames'][dashDict['displayName']]=''
+
+    fileNameParts = jsonFile.split('.')
+    check_json_file_name(jsonFile, fileNameParts)
+
+    pathMap[directory]['jsonIds'][fileNameParts[0]]=''
+
+def check_metadata_entries(path, sample_dashboard):
+  '''Check if metadata file fields are as expected'''
+  required_fields = {"category", "id", "display_name", "description"}
+  missing_fields = required_fields - sample_dashboard.keys()
+  if missing_fields:
+    raise MetadataFormattingError("{} missing {}".format(path, missing_fields))
+
+def check_metadata(directory, pathMap):
+  '''Checks metadata file in directory for correct fields and matching data in 
+  json files within the same directory'''
+  path = os.path.join('.', 'dashboards', directory, 'metadata.yaml')
+
+  with open(path) as f:
+    data = yaml.safe_load(f)
+  sample_dashboards = data.get("sample_dashboards")
+  if not sample_dashboards:
+    raise MetadataFormattingError("sample_dashboards not defined in {}".format(path))
+  for sample_dashboard in sample_dashboards:
+    # Call function to check metadata file fields
+    check_metadata_entries(path, sample_dashboard)
+
+    dashboardId = sample_dashboard.get('id')
+    dashboardName = sample_dashboard.get('display_name')
+
+    pathMap[directory]['metadataIds'][dashboardId]=''
+    pathMap[directory]['metadataNames'][dashboardName]=''
+
+    if not dashboardId in pathMap[directory]['jsonIds'].keys():
+      raise MetadataInJsonMisMatchError("{} does not have a matching json file".format(dashboardId))
+    
+    if not dashboardName in pathMap[directory]['jsonNames'].keys():
+      raise MetadataInJsonMisMatchError("{} does not have a matching json file".format(dashboardName))
+
+def check_json_in_metadata(directory, pathMap):
+  '''Checks json file data against metadata file of the same directory'''
```

Comment:
Similar to my comment above, this could be more specific

```
Make following assertions about json template files:
* Template file name appears in metadata.yaml file
* Template display name matches corresponding id in metadata.yaml file
```

Reply:
done.